### PR TITLE
AA-629 added editable patient relationships

### PR DESCRIPTION
**Pull request by ellymakuba**

### CODE REVIEW

**Comment by ellymakuba** [comment]
Test build link "https://test1.ampath.or.ke:8443/test-builds/AA-629/",
test patient: 821795398-6

File: app/scripts/patient-dashboard/directives/patient-relationship.directive.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  }
+  $scope.setPatientRelationship=setPatientRelationship;
+  function setPatientRelationship(personA,relationshipTypeUUid,personB){
+    if(angular.isDefined(relationshipTypeUUid) && $scope.patient.uuid()!==personB){
+    var relationshipExists=false;
+    _.find($scope.patientRelationships.relationships,function(relationship){
+      console.log("relationship in find loop is ",relationship.uuId());
+      if(personB==relationship.personUuId()) {
+        relationshipExists=true;
+        var patientRelationshipPayload={
```
Comment (nkimaina) [inline app/scripts/patient-dashboard/directives/patient-relationship.directive.js:49]: For readability, you want to take this to a function whose purpose is to create payload for relationships.

File: app/scripts/patient-dashboard/directives/patient-relationship.directive.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+      console.log("patient uuid clicked id ",patientUuid);
+    OpenmrsRestService.getPatientService().getPatientByUuid({ uuid: patientUuid },
+           function(data) {
+             $rootScope.broadcastPatient = data;
+             $state.go('patient', { uuid: patientUuid });
+           }
+      );
+  }
+  $scope.setPatientRelationship=setPatientRelationship;
+  function setPatientRelationship(personA,relationshipTypeUUid,personB){
```
Comment (nkimaina) [inline app/scripts/patient-dashboard/directives/patient-relationship.directive.js:41]: Add one line between functions to make it readable. Also this function is long and there exists opportunities to reuse bits of your code. For e.g. you could take the success and error callbacks to their own functions, and avoid the anonymous functions.

File: app/scripts/patient-dashboard/directives/patient-relationship.directive.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      },
+      function(error){
+        console.error("error response ",error);
+        var dlgError=dialogs.error('Error','System encountered an error while deleting the relationship');
+        dlgError.result.then(function(btnError){});
+      });
+  });
+  $scope.searchString = '';
+}
+$scope.setRelationship=setRelationship;
```
Comment (nkimaina) [inline app/scripts/patient-dashboard/directives/patient-relationship.directive.js]: This line belongs to the declaration section of the controller, up-top.

File: app/scripts/patient-dashboard/directives/patient-relationship.directive.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+            displayErrorDialog('Saving Error','The system encountered an error while saving the relatinship');
+          });
+      }
+    });
+    console.log("relationshipExists = ",relationshipExists);
+    if(!relationshipExists){
+      var patientRelationshipPayload=getPatientRelationshipPayload(personA,relationshipTypeUUid,personB);
+      OpenmrsRestService.getPatientRelationshipTypeService().setPatientRelationship(patientRelationshipPayload,
+      function(success){
+        console.log("successfull POST request ",success);
```
Comment (nkimaina) [inline app/scripts/patient-dashboard/directives/patient-relationship.directive.js]: missing call to success here

**Pull request merged**